bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -347,14 +356,37 @@ private void doMutation(final Mutation mutation) throws RetriesExhaustedWithDeta
       // requirements are achieved.
       sequenceId = sizeManager.registerOperationWithHeapSize(mutation.heapSize());
     } catch (InterruptedException e) {
+      synchronized (globalExceptions) {
+        // Add the exception to the list of global exceptions and immediately throw a
+        // RetriesExhaustedWithDetailsException.
+        addGlobalException(mutation, e);
+        handleExceptions();
+      }
       Thread.currentThread().interrupt();
     }
     AccountingFutureCallback callback = new AccountingFutureCallback(mutation, sequenceId);
 
-    // TODO: Consider adding the callback in another executor for the blocking call. There are some
-    // concerns running this callback on the rpc threadpool. This callback requires locks on
-    // resources. If they cannot be acquired, the callback would block future completions.
-    Futures.addCallback(batchExecutor.issueRequest(mutation), callback);
+    ListenableFuture<? extends GeneratedMessage> request = null;
+    try {
+      request = batchExecutor.issueRequest(mutation);
```

Comment:
Is this raising an UnsupportedOperationException? (that is the one I found reading). It is a pity that it is an unchecked exception, otherwise it would have been easier to catch this. :/ Maybe it should be added to issueRequest and OperationAdapter.adapt?